tests/test_auth.py

```diff
@@ -13,8 +13,8 @@
 
 AUTH_URL = f"{BACKEND_SELECTOR_MOCK.base_url}/oauth/token"
 AUTH_DATA = {
-    "client_id": BACKEND_SELECTOR_MOCK.client_id,
-    "client_secret": BACKEND_SELECTOR_MOCK.client_secret,
+    "client_id": BACKEND_SELECTOR_MOCK.credentials[0]["client_id"],
```

Comment:
Can you also add tests for the new features, please?
- shared appliances
- ensuring that the auth goes trough the two client credentials?